vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/java/com/vaadin/flow/component/spreadsheet/Spreadsheet.java

```diff
@@ -4804,15 +4812,15 @@ private boolean isCellVisible(int row, int col) {
     }
 
     private void registerPopupButton(PopupButton button) {
-        if (popupButtonsEnabled) {
-            attachedPopupButtons.add(button);
-            registerCustomComponent(button);
-        }
+        attachedPopupButtons.add(button);
+        registerCustomComponent(button);
```

Comment:
The `registerCustomComponent` needs some improvements. As it is being called for every `PopupButton` being created and it sends the list of `PopupButtonState` to the client. It actually only needs to be called once, after all the popups have been created.